tests/test_simulations.py

```diff
@@ -0,0 +1,280 @@
+    self.start_time = datetime(2020,8,15,5,0,0)
+    self.end_time = datetime(2020,8,15,5,0,10)
+    test_simulation._start_time = self.start_time
+    test_simulation._end_time = self.end_time
+
+    open_mock = mock_open()
+    with patch("geobeam.simulations.open", open_mock, create=True):
+      open_mock.return_value = MagicMock()
+      test_simulation.log_run(mock_logfile)
+
+    mock_csv.writer.assert_called_once()
+    fields = ["simulation_type", "file_path", "run_duration", "gain", "start_time", "end_time"]
+    values = ["DynamicSimulation", self.file_path, self.run_duration, self.gain, "2020-08-15T05:00:00", "2020-08-15T05:00:10"]
+    gps_fields = ["time_from_zero", "x", "y", "z"]
+    csv_calls = [call(fields), call(values), call(gps_fields)]
+    mock_csv.writer().writerow.assert_has_calls(csv_calls)
+
+    open_mock.assert_called_with(self.file_path, "r")
+    self.assertEqual(mock_logfile.write.call_args_list[0], call('\n'))
+    # check for 1 blank line call and 100 copied lines for 10 seconds of data
+    self.assertEqual(mock_logfile.write.call_count,101)
+
+  @patch('geobeam.simulations.subprocess')
+  def test_create_blade_GPS_process(self, mock_subprocess):
+    mock_subprocess.Popen = Mock()
+    mock_subprocess.PIPE = Mock()
+
+    commands = [["./run_bladerfGPS.sh", "-T", "now"],
+                ["./run_bladerfGPS.sh", "-T", "now", "-d", "20"],
+                ["./run_bladerfGPS.sh", "-T", "now", "-a", "-2"],
+                ["./run_bladerfGPS.sh", "-T", "now", "-u", "test/path"],
+                ["./run_bladerfGPS.sh", "-T", "now", "-d", "20", "-a", "-2"],
+                ["./run_bladerfGPS.sh", "-T", "now", "-d", "20", "-a", "-2", "-l", "27.12345,-37.45678"],
+                ["./run_bladerfGPS.sh", "-T", "now", "-d", "20", "-a", "-2", "-u", "test/path"],
+                ["./run_bladerfGPS.sh", "-T", "now", "-d", "20", "-a", "-2", "-l", "27.12345,-37.45678"]]
+
+    results = [geobeam.simulations.create_bladeGPS_process(),
+               geobeam.simulations.create_bladeGPS_process(run_duration=20),
+               geobeam.simulations.create_bladeGPS_process(gain=-2),
+               geobeam.simulations.create_bladeGPS_process(dynamic_file_path="test/path"),
+               geobeam.simulations.create_bladeGPS_process(run_duration=20, gain=-2),
+               geobeam.simulations.create_bladeGPS_process(run_duration=20, gain=-2,
+                                                           location="27.12345,-37.45678"),
+               geobeam.simulations.create_bladeGPS_process(run_duration=20, gain=-2, 
+                                                           dynamic_file_path="test/path"),
+               geobeam.simulations.create_bladeGPS_process(run_duration=20, gain=-2,
+                                                           location="27.12345,-37.45678",
+                                                           dynamic_file_path="test/path")]
```

Comment:
Typically this is something we'd use a parameterized test for (https://pypi.org/project/parameterized/) so that one failed assert doesn't hide the others.